backend.native/tests/runtime/concurrent/shared0.kt

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package runtime.concurrent.shared0
+
+import kotlin.test.*
+
+import kotlin.native.concurrent.*
+import kotlin.native.internal.GC
+import kotlin.native.ref.WeakReference
+
+class A(var a: Int)
+
+val global1: SharedRef<A> = SharedRef(A(3))
+
+@Test fun testGlobal() {
+    assertEquals(3, global1.get().a)
+
+    val worker = Worker.start()
+    val future = worker.execute(TransferMode.SAFE, {}) {
+        global1
+    }
+
+    val value = future.result
+    assertEquals(3, value.get().a)
+    worker.requestTermination().result
+}
+
+val global2: SharedRef<A> = SharedRef(A(3))
+
+@Test fun testGlobalDenyAccessOnWorker() {
+    assertEquals(3, global2.get().a)
+
+    val worker = Worker.start()
+    val future = worker.execute(TransferMode.SAFE, {}) {
+        var result = 0
+        assertFailsWith<IncorrectDereferenceException> {
+            result = global2.get().a
```

Comment:
This actually checks that either `global2`, `.get()` or `.a` throws an exception.
It would make more sense to test that `.get()` throws.